common/prometheus-server/templates/thanos/storeendpoint.yaml

```diff
@@ -0,0 +1,15 @@
+{{- if and .Values.thanos.enabled ( eq .Values.thanos.objectStorageConfig.enabled false ) }}
+{{- $root := . }}
+{{- range $name := coalesce .Values.names .Values.global.targets (list .Values.name) }}
+{{- range $thanos := $.Values.thanos.sidecarDiscovery }} 
+---
+apiVersion: monitoring.banzaicloud.io/v1alpha1
+kind: StoreEndpoint
+metadata:
+  name: {{ include "prometheus.name" (list $name $root) }}-to-{{ $thanos.thanosName }}
+spec:
+  url: {{ include "prometheus.name" (list $name $root) }}.{{ $thanos.namespace }}.svc.kubernetes.{{- required ".Values.global.region missing" $root.Values.global.region -}}.{{- required ".Values.global.domain missing" $root.Values.global.domain }}:10901
```

Comment:
Please add the default cluster domain as well: `cluster.local`. Or make it customisable via values.

Reply:
We can also use this pattern w/o port:
`dnssrvnoa+_grpc._tcp.{{ include "prometheus.name" (list $name $root) }}.{{ $thanos.namespace }}.svc.{{ .Values.clusterDomain }}`